R/nested_ifelse_linter.R

```diff
@@ -9,9 +9,52 @@
 #'      is exacerbated for nested calls
 #'
 #' Users can instead rely on a more readable alternative modeled after SQL
-#'   CASE WHEN statements, such as `data.table::fcase()` or `dplyr::case_when()`,
-#'   or use a look-up-and-merge approach (build a mapping table between values
-#'   and outputs and merge this to the input).
+#'   CASE WHEN statements.
+#'
+#' Let's say this is our original code:
+#'
+#' ```r
+#' ifelse(
+#'   x == "a",
+#'   2L,
+#'   ifelse(x == "b", 3L, 1L)
+#' )
+#' ```
+#'
+#' Here are a few ways to avoid nesting and make the code more readable:
+#'
+#'   - Use `data.table::fcase()`
+#'
+#'     ```r
+#'     data.table::fcase(
+#'       x == "a", 2L,
+#'       x == "b", 3L,
+#'       default = 1L
+#'     )
+#'     ```
+#'
+#'   - Use `dplyr::case_when()`
```

Comment:
This particular case should actually be handled by `dplyr::recode(x, "a" = 2L,  "b" = 3L, .default = 1L)`

Reply:
`recode()` has been superseded in favour of `case_match()`: https://dplyr.tidyverse.org/reference/recode.html

Reply:
Oh, thanks for the info. Then with `case_match()` accordingly.